### PR TITLE
Harbor upgrading for 2.2

### PR DESCRIPTION
**Pull request by ninjadq**
1. if san not exists then remove that cert, prepare will regenerate one
2. Add upgrade script

Signed-off-by: DQ <dengq@vmware.com>

### CODE REVIEW

File: make/photon/prepare/migrations/__init__.py [hunk 1]
```diff
@@ -2,4 +2,4 @@
 
 MIGRATION_BASE_DIR = os.path.dirname(__file__)
 
-accept_versions = {'1.9.0', '1.10.0', '2.0.0', '2.1.0'}
\ No newline at end of file
+accept_versions = {'1.9.0', '1.10.0', '2.0.0', '2.1.0', '2.2.0'}
```
Comment (reasonerjt) [inline make/photon/prepare/migrations/__init__.py:5]: I don't think we need to keep adding versions as we only support n-2 upgrading path?

File: make/photon/prepare/migrations/version_2_2_0/harbor.yml.jinja [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+  http_proxy:
+  https_proxy:
+  no_proxy:
+  components:
+    - core
+    - jobservice
+    - trivy
+{% endif %}
+
+{% if metric is defined %}
```
Comment (reasonerjt) [inline make/photon/prepare/migrations/version_2_2_0/harbor.yml.jinja:425]: will there be `metric` in 2.1.0?

**Review by wy65701436 — APPROVED** [review]
lgtm

File: make/photon/prepare/migrations/__init__.py (hunk 1 above)
Comment (ninjadq) [reply make/photon/prepare/migrations/__init__.py:5]: This is the input config's version list which our prepare script can handle. Even we only maintain n-2 versions, we shouldn't block earlier versions to upgrade.

Comment (reasonerjt) [reply make/photon/prepare/migrations/__init__.py:5]: Fair enough.

**Review by reasonerjt — APPROVED** [review]

**Pull request merged**